department-of-veterans-affairs/caseflow
# APPEALS-30938 | Add DecisionReviewType condition for `ReportPage`

Jira: https://jira.devops.va.gov/browse/APPEALS-30938

Main Jira effort: https://jira.devops.va.gov/browse/APPEALS-26499

# Description
- Add the `DecisionReviewType` condition field inputs for the Generate Task Report effort.
- Refactor `ConditionContent` rendering
   1. Add a "middle content" section that is conditionally displayed.
This displays the "includes" text for all but the `daysWaiting`
condition. As a result of this some css changes were made to correctly
display this text.

   2. Remove the getConditionContent() function and instead change that
to a memoized conditionContent variable. I can't find an article on it
but I was taught to avoid calling functions that render React
components within jsx. I believe it caused rendering/state issues but
idk I can't find anything on it so who knows. At the least this should
have better performance using useMemo

## Acceptance Criteria
- [ ] There should be an additional condition option: Decision Review Type 
- [ ] The checkbox options for Decision Review Type are Higher-Level Reviews and Supplemental Claims
- [ ] The data gets correctly set and can be seen in the console log on submit of the form.


## Testing Plan
1. Log in as VHAADMIN1
2. Go to `decision_reviews/vha/report`
3. Click Add Condition
4. Select Decision Review Type
5. See the checkbox fields and interact with them
6. Click submit and check the dev tools console to see if the correct data was logged

## Design Note:
The implemented changes are slightly different than the designs in Figma. The designs in Figma assume a larger width of the main app container than what currently exists in our app. As a result the elements becomes squished when trying to match the figma exactly. Instead I have taken some design liberties to wrap certain elements when the screen width gets too small. Although this is not in the designs I presume this will be fine. Screen share can be seen below:



https://github.com/department-of-veterans-affairs/caseflow/assets/51007432/6308dd71-4515-413d-ba0a-b3772a97d878
